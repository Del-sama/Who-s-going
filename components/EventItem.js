import React, { Component } from 'react'
import { View,
         Text,
         TouchableHighlight,
         StyleSheet} from 'react-native'
import { Card } from 'react-native-elements'
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';


const TouchableMaterialIcons = ({ name, size, onPress }) => {
    return (
        <TouchableHighlight>
            <MaterialIcons 
                name={name} 
                size={size || 24} 
                onPress={onPress}
            />
        </TouchableHighlight>
    )
}

const EventItem = ({ title, date, selectHandler }) => {
    return (
        <Card containerStyle={styles.containerStyle}>
            <View style={styles.cardContent}>
                <View style={styles.eventDescription}>
                    <Text style={styles.titleText}>{title}</Text> 
                    <Text style={styles.dateStyle}>Date: {date}</Text>
                </View>
            </View>
        </Card>
    )
}

export default EventItem;


const styles = StyleSheet.create({
    cardContent: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    titleText: {
        fontSize: 18,
    },
    eventDescription: {
        flex: 8,
        flexDirection: 'column',
        height: 50,
        width: 80,
        justifyContent: 'space-between',
    },
    containerStyle: {
        height: 80,
        width: 315,
    },
    dateStyle: {
        fontSize: 14
    }
})
