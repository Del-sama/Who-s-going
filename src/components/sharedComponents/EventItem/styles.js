import { StyleSheet } from 'react-native';

export default styles = StyleSheet.create({
    cardContent: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    titleText: {
        fontSize: 18,
    },
    eventDescription: {
        flex: 8,
        flexDirection: 'column',
        height: 50,
        width: 80,
        justifyContent: 'space-between',
    },
    containerStyle: {
        height: 80,
        width: 315,
    },
    dateStyle: {
        fontSize: 14
    }
})
