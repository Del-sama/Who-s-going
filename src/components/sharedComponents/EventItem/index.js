import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { Card } from 'react-native-elements';
import styles from './styles';


const EventItem = ({ title, date, onPress }) => (
  <TouchableOpacity
    onPress={onPress}
  >
    <Card containerStyle={styles.containerStyle}>
      <View style={styles.cardContent}>
        <View style={styles.eventDescription}>
          <Text
            style={styles.titleText}
            numberOfLines={1}
          >
            {title}
          </Text>
          <Text style={styles.dateStyle}>
            Date:
            {date}
          </Text>
        </View>
      </View>
    </Card>
  </TouchableOpacity>

);

export default EventItem;
