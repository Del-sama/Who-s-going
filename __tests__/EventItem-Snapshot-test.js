import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventItem from '../src/components/sharedComponents/EventItem';

it('renders correctly', () => {
  const tree = renderer.create(
    <EventItem />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
