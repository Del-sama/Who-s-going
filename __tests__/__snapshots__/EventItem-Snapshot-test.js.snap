// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      Object {
        "height": 80,
        "width": 315,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 8,
            "flexDirection": "column",
            "height": 50,
            "justifyContent": "space-between",
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          Date: 
        </Text>
      </View>
    </View>
  </View>
</View>
`;
